File: 151001/Maliauka/src/main/java/by/bsuir/dc/features/news/dto/NewsRequestDto.java
package by.bsuir.dc.features.news.dto;

import by.bsuir.dc.features.news.News;
import jakarta.validation.constraints.*;

import java.io.Serializable;
import java.time.Instant;

/**
 * DTO for {@link News}
 */
public record NewsRequestDto(
        @NotBlank @Size(min = 2, max = 64) String title,
        @NotBlank @Size(min = 4, max = 2048) String content,
        @Past Instant created,
        @Past Instant modified,
        @Min(1) @Max(Long.MAX_VALUE) Long editorId
) implements Serializable {}

File: 151002/Haiduk/RV/RV/Lab5/Lab5.Publisher/DTO/RequestDTO/NewsRequestDto.cs
namespace Lab5.Publisher.DTO.RequestDTO;

public class NewsRequestDto
{
    public long Id { get; set; }
    public long CreatorId { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }
}